tests/io/eager/test_read.py

```diff
@@ -159,3 +159,30 @@ def test_write_and_read_default_table_name(store_session):
         "test_default_table_name", store_session
     )
     pd.testing.assert_frame_equal(df_write, df_read_as_dfs[0]["table"])
+
+
+def test_read_or_predicates(store_session):
+    # https://github.com/JDASoftwareGroup/kartothek/issues/295
```

Comment:
This test includes the example from the original description, where the first `read_table` would raise a `Value Error: empty Predicates`. I might have fixed this behaviour accidentally, by focusing on the correct "OR" evaluation during dispatch.

I have included a test for `filter_df_from_predicates`, which works as intended (cc @fjetter), currently looking into the filtering here: https://github.com/JDASoftwareGroup/kartothek/blob/master/kartothek/io_components/metapartition.py#L604
🔝 This can return an empty list as a result for `filtered_predicate`, which leads to the raise of the Value Error later.